store/sqlstore/post_store.go

```diff
@@ -2956,6 +2960,20 @@ func (s *SqlPostStore) savePostsPriority(transaction *sqlxTxWrapper, posts []*mo
 	return nil
 }
 
+func (s *SqlPostStore) savePostsPersistentNotifications(transaction *sqlxTxWrapper, posts []*model.Post) error {
+	for _, post := range posts {
+		if priority := post.GetPriority(); priority != nil && priority.PersistentNotifications != nil && *priority.PersistentNotifications {
+			if _, err := transaction.NamedExec(`INSERT INTO PersistentNotifications (PostId, CreateAt, LastSentAt, DeleteAt, SentCount) VALUES (:PostId, :CreateAt, :LastSentAt, :DeleteAt, :SentCount)`, &model.PostPersistentNotifications{
+				PostId:   post.Id,
+				CreateAt: post.CreateAt,
+			}); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
```

Comment:
> If I move it, then the transaction param will also need to be moved which will require a SQL import stmt. into the generic store.go file, which doesn't seem right to me.

For the `transaction` param, you can use the `sqlxExecutor` interface: https://github.com/mattermost/mattermost-server/blob/e055df27e87d22893601bc298949ee7ae5bc3f89/store/sqlstore/sqlx_wrapper.go#L42-L56, making the called code agnostic.

> Also, since it's a part of the single post-insertion transaction, we can keep it here along with the other funcs similar to the thread and priority updates.

I'm only 2/5 on this, but we've ran into issues where thread logic wasn't properly updated in all the places that touched that schema. I think it makes sense to have stores own writing to their own tables.

Reply:
Actually I just checked again and there is already a new import stmt. here `"database/sql"`. So, I guess the store isn't that generic anymore.

Will move the function out.

Reply:
> Tried, can't import it. It creates a cycle between the two packages. sqlstore -> store -> sqlstore

Huh, weird: can you elaborate on what new imports are required? The post and persistent notification stores and `sqlxExecutor` interface are already in the same package.

Agreed re: cache layer in general, but that doesn't apply to a write method anyway.

Reply:
Yes, all 3 are in the same package `sqlstore`. But ATM there is no reference of any model or function in `store.go` file which is in `store` package. The `store.go` file has interfaces which references only `model` but without referencing their implementations in `sqlstore`. 

So, now if I were to reference the `sqlxExecutor` from `sqlstore` it will create the cycle.